### PR TITLE
Shellcheck

### LINKED ISSUE #11: Add action-shellcheck
Add `action-shellcheck` to lint shell scripts.

Link:
https://github.com/reviewdog/action-shellcheck

### PR DESCRIPTION
**Pull request by vterdunov**
* Add shellcheck to CI.
* Use bash for more safety scripting.

fix #11 
shell doesn't support `[[`
image

### CODE REVIEW

**Review by mgrachev — APPROVED** [review]

**Comment by review-dog** [comment]
Hi, @vterdunov! We merged your PR to reviewdog! 🐶
Thank you for your contribution! ✨

**We just invited you to join the @reviewdog organization on GitHub.**
Accept the invite by visiting https://github.com/orgs/reviewdog/invitation.
By joining the team, you'll be a part of reviewdog community and can help the maintainance of reviewdog.

Thanks again!

File: script.sh [hunk 1]
```diff
@@ -21,29 +25,28 @@ if [[ $INPUT_REEK_VERSION = "gemfile" ]]; then
...
       printf 'Gemfile.lock not found. The latest version will be installed.'
   fi
-  else
-    # set desired reek version
-    REEK_VERSION=$INPUT_REEK_VERSION
 fi
 
-gem install -N reek $(version $REEK_VERSION)
+# shellcheck disable=SC2046,SC2086
+sudo gem install --no-document reek $(version $REEK_VERSION)
```
Comment (mrexox) [inline script.sh:43]: This change is breaking. Why using sudo here?

Comment (vterdunov) [reply script.sh:43]: Hi. Could you provide a bit more information how this change breaks your CI? sudo is used to system-wide install reek.

Comment (mrexox) [reply script.sh:43]: It does not work on hosted runners. I think, using `ruby/setup-ruby` is overkill here, where the only need is to run linters.
`vterdunov/action-reek@do-not-use-sudo` worked fine :+1:

**Pull request merged**